st2api/st2api/controllers/v1/keyvalue.py

```diff
@@ -161,7 +172,18 @@ def get_all(
         self._validate_decrypt_query_parameter(
             decrypt=decrypt, scope=scope, requester_user=requester_user
         )
-
+        if decrypt:
+            LOG.audit(
+                "User %s decrypted the value %s ",
+                user,
+                name,
```

Comment:
Since this is `get_all`, there isn't a key `name` here (that's what lint is complaining about). The wording might need to change a bit to reflect that it is more than one key.

Reply:
yeah your right, I'm wondering if a decryption event occurs in the get_all function? I can not find the variable or track the function at the point where a decryption occurs?

Reply:
OK. I think we can add a couple of audit log entries in `get_all()`.
- around line 246 for system scope
```python
        if scope in [ALL_SCOPE, SYSTEM_SCOPE, FULL_SYSTEM_SCOPE]:
            decrypted_keys = []
            # ...
            if has_system_role:
                ...
                kvp_apis_system.extend(items.json or [])
                if decrypt and items.json:
                    decrypted_keys.extend(kvp_api.name for kvp_api in items.json if kvp_api.secret)
                for key in get_all_system_kvp_names_for_user(current_user):
                        ...
                        kvp_apis_system.append(item)
                    except Exception as e:
                        LOG.error("Unable to get key %s: %s", key, str(e))
                        continue
                    if decrypt and item.secret:
                        decrypted_keys.append(key)
            if decrypted_keys:
                LOG.audit(
                    ...
                    extra={... "scope": FULL_SYSTEM_SCOPE, ...}
                )
```

- around line 267 for user scope
```python
        if scope in [ALL_SCOPE, USER_SCOPE, FULL_USER_SCOPE]:
            ...
            kvp_apis_user.extend(items.json)
            if decrypt and items.json:
                decrypted_keys = [kvp_api.name for kvp_api in items.json if kvp_api.secret]
                if decrypted_keys:
                    LOG.audit(
                        ...
                        extra={... "scope": FULL_USER_SCOPE, ...}
                    )
```

What do you think of having one system and one user log entry for all of the decrypted keys like this?

Reply:
That sounds great, Ive added these new updates now

Reply:
I think line numbers for are confusing when editing a file. LOL

Looks like you have repo audit lids for system scope but missed the user scope, provable because the lounge numbers changed with moving lines around.

Reply:
ahaha yeah it really is. OKay I've added the changes now let me know what you think.